tutorials/plot_background_statistics.py

```diff
@@ -0,0 +1,566 @@
+# The first three columns show the parametric and non-parametric statistics
+# that are not corrected for multiple comparisons:
+#
+# - **t-test** has jagged edges.
+# - **"hat" variance correction** of the t-test has reduced peaky edges,
+#   correcting for sharpness in the statistic driven by low-variance voxels.
+# - **non-parametric resampling test** is very similar to the t-test, as
+#   the data are parametric.
+#
+# The next three columns show multiple comparison corrections of the
+# mass univariate tests (parametric and non-parametric). These
+# too conservatively correct for multiple comparisons because neighboring
+# voxels in our data are correlated:
+#
+# - **Bonferroni correction** eliminates any significant activity.
+# - **FDR correction** is less conservative than Bonferroni.
+# - **Permutation test with a maximum statistic** also eliminates any
+#   significant activity.
+#
+# The final four columns show the non-parametric, cluster-based permutation
+# tests with a maximum statistic:
+#
+# - **Standard clustering** identifies the correct region. However, the whole
+#   area must be declared significant, so no peak analysis can be done.
+#   Also, the peak is broad.
+# - **Clustering with "hat"** tightens the estimate of significant activity.
+# - **Clustering with TFCE** allows analyzing each significant point
+#   independently, but still has a broadened estimate.
+# - **Clustering with TFCE and "hat"** tightens the area declared
+#   significant (again FWER corrected).
+#
+# Statistical functions in MNE
+# ----------------------------
+# MNE provides several convenience parametric testing functions that can be
+# used in conjunction with the non-parametric clustering methods. However,
+# the set of functions we provide is not meant to be exhaustive.
+#
+# If the univariate statistical contrast of interest to you is not listed
+# here (e.g., interaction term in an unbalanced ANOVA), consider checking
+# out the :mod:`statsmodels` package. It offers many functions for computing
+# statistical contrasts, e.g., :func:`statsmodels.stats.anova.anova_lm`.
+# To use these functions in clustering:
+#
+# 1. Determine which statistical test you would use in a univariate context
+#    (e.g., to compute your contrast of interest if there is only a single
+#    output, like reaction times).
+# 2. Wrap the call to that function within a function that takes an input of
+#    the same shape that is expected by your clustering function,
+#    and returns an array of the same shape without the "samples" dimension
+#    (e.g., :func:`mne.stats.permutation_cluster_1samp_test` takes an array
+#    of shape ``(n_samples, p, q)`` and returns an array of shape ``(p, q)``).
+# 3. Pass this wrapped function to the ``stat_fun`` argument to the clustering
+#    function.
+# 4. Set an appropriate ``threshold`` value (float or dict) based on the
+#    values your statistical contrast function returns.
```

Comment:
Maybe I'm missing something, but this talks about tests, but isn't this more about different methods for supplying test *statistics*?

Reply:
yes good call